xload/providers/cached/cache.go

```diff
@@ -0,0 +1,75 @@
+package cached
+
+import (
+	"sync"
+	"time"
+)
+
+// Cacher is the interface for custom cache implementations.
+//
+//go:generate mockery --name Cacher --structname MockCache --filename mock_test.go --outpkg cached --output .
+type Cacher interface {
+	Get(key string) (string, error)
+	Set(key, value string, ttl time.Duration) error
+}
+
+// MapCache is a simple cache implementation using a map.
+type MapCache struct {
+	m   map[string]string
+	ttl map[string]time.Time
+
+	now func() time.Time
+	mu  sync.Mutex
```

Comment:
Why not use a `sync.Map` with a wrapper struct?
Equivalent to `map[string]*mv`.

```go
type mv struct {
	val string
	ttl time.Time
}
```

Reply:
Makes sense. Changed it